acinclude.m4

```diff
@@ -602,6 +602,7 @@ AC_DEFUN([AM_ICONV],
   ])
   if test "$am_cv_func_iconv" = yes; then
     AC_DEFINE(HAVE_ICONV, 1, [Define if you have the iconv() function.])
+    AC_LANG_PUSH(C++)
```

Comment:
I think it would be better to do this outside the call to `AM_ICONV` in our `configure.in` as this is a 3rd party macro (we should arguably put `iconv.m4` under `build/aclocal` instead of having it here at all) and modifying it will make updating it later more difficult.

Could you please do it like this and, most importantly, check that it indeed works? TIA!